quickstarts/microsoft.batch/batch-pool-no-public-ip/main.bicep

```diff
@@ -0,0 +1,195 @@
+          }
+        ]
+      }
+      nodeManagementAccess: {
+        defaultAction: 'Deny'
+      }
+    }
+  }
+}
+
+resource privateDnsZone 'Microsoft.Network/privateDnsZones@2020-06-01' = {
+  name: privateDnsZoneName
+  location: 'global'
+}
+
+resource nsg 'Microsoft.Network/networkSecurityGroups@2022-01-01' = {
+  name: nsgName
+  location: location
+  properties: {
+    securityRules: [
+      {
+        name: 'DenyInternetOutbound'
+        properties: {
+          access: 'Deny'
+          protocol: '*'
+          direction: 'Outbound'
+          sourceAddressPrefix: '*'
+          sourcePortRange: '*'
+          destinationAddressPrefix: 'Internet'
+          destinationPortRange: '*'
+          priority: 200
+        }
+      }
+    ]
+  }
+}
+
+resource vnet 'Microsoft.Network/virtualNetworks@2022-01-01' = {
+  name: vnetName
+  location: location
+  dependsOn: [
+    nsg
+  ]
+  properties: {
+    addressSpace: {
+      addressPrefixes: [
+        '10.0.0.0/16'
+      ]
+    }
+    subnets: [
+      {
+        name: subnetName
+        properties: {
+          addressPrefix: '10.0.0.0/24'
+          networkSecurityGroup: {
+            id: resourceId('Microsoft.Network/networkSecurityGroups', nsgName)
+          }
+          privateEndpointNetworkPolicies: 'Disabled'
+        }
+      }
+    ]
+  }
+}
+
+resource vnetDnsZoneLink 'Microsoft.Network/privateDnsZones/virtualNetworkLinks@2020-06-01' = {
+  name: '${privateDnsZoneName}/${vnetName}'
+  location: 'global'
+  dependsOn: [
+    privateDnsZone
+    vnet
+  ]
+  properties: {
+    registrationEnabled: false
+    virtualNetwork: {
+      id: resourceId('Microsoft.Network/virtualNetworks', vnetName)
+    }
+  }
+}
+
+resource nodeManagementPrivateEndpoint 'Microsoft.Network/privateEndpoints@2022-01-01' = {
+  name: nodeManagementPrivateEndpointName
+  location: location
+  dependsOn: [
+    batchAccount
+    vnet
+  ]
+  properties: {
+    privateLinkServiceConnections: [
+      {
+        name: nodeManagementPrivateEndpointName
+        properties: {
+          privateLinkServiceId: resourceId('Microsoft.Batch/batchAccounts', accountName)
+          groupIds: [
+            'nodeManagement'
+          ]
+        }
+      }
+    ]
+    subnet: {
+      id: resourceId('Microsoft.Network/virtualNetworks/subnets', vnetName, subnetName)
+    }
+  }
+}
+
+resource privateEndpointDnsIntegration 'Microsoft.Network/privateEndpoints/privateDnsZoneGroups@2020-11-01' = {
+  name: '${nodeManagementPrivateEndpointName}/default'
+  dependsOn: [
+    nodeManagementPrivateEndpoint
+    privateDnsZone
+  ]
+  properties: {
+    privateDnsZoneConfigs: [
+      {
+        name: 'private-dns-zone-integration'
+        properties: {
+          privateDnsZoneId: resourceId('Microsoft.Network/privateDnsZones', privateDnsZoneName)
+        }
+      }
+    ]
+  }
+}
+
+resource batchPool 'Microsoft.Batch/batchAccounts/pools@2022-06-01' = {
+  name: '${accountName}/${poolName}'
+  dependsOn: [
```

Comment:
remove dependsOn

Reply:
Keep dependency on private endpoint as the pool requires it to work normally.